### ISSUE TITLE
Provide a way to not require space after 'new' for shorthand initializers

### ISSUE THREAD

**Issue by jez9999** [DotNetAnalyzers/StyleCopAnalyzers#3558]
When converting to shorthand `new` initializer:

```
public static readonly List<Foo> _directions = new List<Foo> {
...
public static readonly List<Foo> _directions = new() {
```

Rule [SA1000](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1000.md) is triggered.  I don't want to disable this entire rule as most of it is desirable, but I also don't agree that space needs to go after `new` in this case (and actually, inserting a space then triggers an `IDE0055` warning, so the two are in conflict).  Please provide a way to configure StyleCop so spaces after `new` in this case aren't required by this rule.
